src/main/java/controller/GameController.java

```diff
@@ -1,33 +1,83 @@
 package controller;
 
-import domain.GameBoard;
+import domain.*;
 import domain.ladder.Ladder;
-import domain.ladder.common.Height;
+import domain.ladder.attribute.Height;
+import domain.common.Name;
 import domain.player.Names;
+import domain.player.Player;
 import domain.player.Players;
+import domain.reward.Result;
+import domain.reward.Rewards;
 import util.RandomDirectionGenerator;
 import view.InputView;
 import view.OutputView;
 
+import java.util.List;
 import java.util.stream.IntStream;
 
 public class GameController {
+    private static final String ALL_COMMAND = "all";
+
     public void execute() {
-        Names names = Names.from(InputView.inputPlayerNames());
+        Players players = createPlayers();
+        Rewards rewards = Rewards.from(InputView.inputRewards(), players.getPlayerCount());
         Height height = new Height(InputView.inputHeight());
-
-        Players players = new Players(names);
         Ladder ladder = new Ladder(height, players.getPlayerCount(), new RandomDirectionGenerator());
-        GameBoard gameBoard = new GameBoard(players, ladder);
+        GameBoard gameBoard = new GameBoard(players, ladder, rewards);
+
+        printGameBoard(gameBoard);
+        processGameResults(gameBoard);
+    }
+
+    private Players createPlayers() {
+        Names names = Names.from(InputView.inputPlayerNames());
+        return new Players(names);
+    }
+
+    private void processGameResults(GameBoard gameBoard) {
+        while (true) {
```

Comment:
아래와 같이 작성하면 break 가 필요없겠군요!
while이 멈추는 조건도 명확하게 보이구요 😉 

```suggestion
String playerName = "";
while(ALL_COMMAND.equals(playerName)) {
    ~~
}
```

Reply:
해당 부분에 대해서 생각을 했지만
결국 String playerName 을 빈 문자열로 초기화 한 후
로직을 실행하는 느낌이라 고민을 하다
```java
    private void processGame(GameBoard gameBoard) {
        final String playerName = InputView.inputResultPlayer();
        if (LadderCommand.isAllCommand(playerName)) {
            processAllPlayers(gameBoard);
            return;
        }
        processSinglePlayer(gameBoard, playerName);
        processGame(gameBoard);
    }
```
해당 코드와 같이 재귀적 함수로 수행을 했습니다.
코드적으로는 더 깔끔하고 , final 선언까지 해서 만족스러운데
찰리의 의견은 어떤가요?? !

Reply:
헉 저는 해당 메서드를 읽었을 때
재귀로 인해 반복된다는 사실을 마지막 라인을 읽어서야 깨달았어요 🤔 

반복된다는 것을 눈치채기 힘든 구조는 아닐까 의심됩니다!

저는 반복문은 `for 또는 while 이 가장 명확하지 않은가` 라고 생각해요!